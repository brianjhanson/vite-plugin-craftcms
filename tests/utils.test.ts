import { expect, test } from "vitest";
import {
  defaultTemplateFunction,
  filenameFromPath,
  formatOutputPath,
  getInputs,
  parseFile,
  replaceAttribute,
} from "../src/utils";

const TEST_HTML = `
<head>
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <link rel="stylesheet" href="./styles/main.scss">
  <link rel="stylesheet" href="./styles/test.scss">
</head>
<body>
  <script type="module" src="./scripts/main.js"></script>
</body>
`;

test("parseFile", () => {
  const { head, body } = parseFile(TEST_HTML);

  expect(head).toHaveLength(2);
  expect(body).toHaveLength(1);
});

test("replaceAttribute", () => {
  const { body } = parseFile(TEST_HTML);

  const result = replaceAttribute(body, "src", "./", "http://localhost/");
  const expected = `<script type="module" src="http://localhost/scripts/main.js"></script>`;
  expect(result.toString()).toContain(expected);
});

test("defaultTemplateFunction production mode", () => {
  const { head, body } = parseFile(TEST_HTML);
  const result = defaultTemplateFunction({ head, body, basePath: "/dist/" });

  expect(result).toMatchSnapshot();
});

test("defaultTemplateFunction development mode", () => {
  const { head, body } = parseFile(TEST_HTML);
  const result = defaultTemplateFunction({
    head,
    body,
    mode: "development",
    proxyUrl: "http://localhost",
  });

  expect(result).toMatchSnapshot();
});

test("formatOutputPath", () => {
  expect(formatOutputPath('./templates/foo.twig')).toEqual("./templates/foo.twig");
  expect(formatOutputPath('./templates/[name].twig', 'foo')).toEqual("./templates/foo.twig");
  expect(formatOutputPath('./templates/vite-[name].twig', 'foo')).toEqual("./templates/vite-foo.twig");
});

test("getInputs", () => {
  expect(getInputs("./src/one.html")).toEqual(["./src/one.html"]);
  expect(getInputs(["./src/one.html", "./src/two.html"])).toEqual(["./src/one.html", "./src/two.html"]);
  expect(getInputs({ one: "./src/one.html", two: "./src/two.html" })).toEqual(["./src/one.html", "./src/two.html"]);
});

test("filenameFromPath", () => {
  [
    './src/foo.html',
    'src/foo.html',
    'foo.html',
  ].forEach((path) => {
    expect(filenameFromPath(path)).toEqual('foo');
  })
});
