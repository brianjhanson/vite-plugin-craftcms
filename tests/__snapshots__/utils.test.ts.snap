// Vitest Snapshot v1

exports[`defaultTemplateFunction development mode 1`] = `
"{%- macro url(path) -%}{{ 'http://localhost' | replace('/\\\\\\\\/$/', '') ~ '/' }}{{ path }}{%- endmacro -%}
{% html at head %}<script type=\\"module\\" src=\\"http://localhost/@vite/client\\"></script><link rel=\\"stylesheet\\" href=\\"http://localhost/src/styles/main.scss\\"><link rel=\\"stylesheet\\" href=\\"http://localhost/src/styles/test.scss\\">{% endhtml %}
{% html at endBody %}<script type=\\"module\\" src=\\"http://localhost/src/scripts/main.js\\"></script>{% endhtml %}"
`;

exports[`defaultTemplateFunction production mode 1`] = `
"{%- macro url(path) -%}{{ '/dist/' | replace('/\\\\\\\\/$/', '') ~ '/' }}{{ path }}{%- endmacro -%}
{% html at head %}<link rel=\\"stylesheet\\" href=\\"/src/styles/main.scss\\"><link rel=\\"stylesheet\\" href=\\"/src/styles/test.scss\\">{% endhtml %}
{% html at endBody %}<script type=\\"module\\" src=\\"/src/scripts/main.js\\"></script>{% endhtml %}"
`;
